### DISCUSSION TITLE
n_jobs > 1 causes entire process to rebuild

### DISCUSSION THREAD

**Discussion by jmrichardson** [alan-turing-institute/sktime#2904] [Q&A] [answered]
Hi,

I am using:

ts = TSFreshFeatureExtractor(default_fc_parameters="minimal", show_warnings=False, n_jobs=4, disable_progressbar=False)

However, it restarts everything in my calling script for each job instead of just starting a new process and bringing copying the environment.  In other words, it's basically restarting the script 4 times for each job.  Am I doing something wrong?  I just want it to use the existing environment variables and begin extracting the features.  If I enable n_jobs=1, everything works fine and the main script doesn't get re-executed.  Thanks for any help.

**Comment by jmrichardson** [comment, marked as answer]
I figured it out, it was because my script didn't begin with:

`if __name__ == "__main__":`

**Reply by fkiraly** [reply]
ah, makes sense. Sometimes it´s not obvious...
